fahrschein/src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -52,16 +52,55 @@
     private final JsonFactory jsonFactory;
     private final ObjectWriter cursorHeaderWriter;
 
+    private final StreamInfoReader streamInfoReader;
+
     private final MetricsCollector metricsCollector;
 
     /*
      * @VisibleForTesting
      */
-    NakadiReader(URI uri, RequestFactory requestFactory, BackoffStrategy backoffStrategy, CursorManager cursorManager, ObjectMapper objectMapper, Set<String> eventNames, Optional<Subscription> subscription, Optional<Lock> lock, Class<T> eventClass, Listener<T> listener) {
-        this(uri, requestFactory, backoffStrategy, cursorManager, eventNames, subscription, lock, new MappingEventReader<>(eventClass, objectMapper), listener, DefaultBatchHandler.INSTANCE, NoMetricsCollector.NO_METRICS_COLLECTOR);
+    NakadiReader(
+            URI uri,
+            RequestFactory requestFactory,
+            BackoffStrategy backoffStrategy,
+            CursorManager cursorManager,
+            ObjectMapper objectMapper,
+            Set<String> eventNames,
+            Optional<Subscription> subscription,
+            Optional<Lock> lock,
+            Class<T> eventClass,
+            Listener<T> listener
+    ) {
+        this(
+                uri,
+                requestFactory,
+                backoffStrategy,
+                cursorManager,
+                eventNames,
+                subscription,
+                lock,
+                new MappingEventReader<>(eventClass, objectMapper),
+                listener,
+                DefaultBatchHandler.INSTANCE,
+                NoMetricsCollector.NO_METRICS_COLLECTOR,
+                StreamInfoReader.getDefault()
+        );
     }
 
-    NakadiReader(URI uri, RequestFactory requestFactory, BackoffStrategy backoffStrategy, CursorManager cursorManager, Set<String> eventNames, Optional<Subscription> subscription, Optional<Lock> lock, EventReader<T> eventReader, Listener<T> listener, BatchHandler batchHandler, final MetricsCollector metricsCollector) {
+    NakadiReader(
+            URI uri,
+            RequestFactory requestFactory,
+            BackoffStrategy backoffStrategy,
+            CursorManager cursorManager,
+            Set<String> eventNames,
+            Optional<Subscription> subscription,
+            Optional<Lock> lock,
+            EventReader<T> eventReader,
+            Listener<T> listener,
+            BatchHandler batchHandler,
+            final MetricsCollector metricsCollector,
+            StreamInfoReader streamInfoReader
```

Comment:
A bit overkill to me - why would someone want to plug their own StreamInfoReader? I think this is only introduced because of testing. Can we find a better way of doing it?